src/orion/core/evc/conflicts.py

```diff
@@ -1553,44 +1553,70 @@ def __init__(self, conflict, new_name):
                 The conflict which is resolved by this resolution.
             new_name: string
                 A new name for the branching experiment. A ValueError is raised if name is already
-                in database.
+                in database with a direct child.
 
             Raises
             ------
             ValueError
-                If name already exists in database for current user.
+                If name already exists in database with a direct child for current user.
 
             """
             super(ExperimentNameConflict.ExperimentNameResolution, self).__init__(conflict)
 
             self.new_name = new_name
-            self.old_name = self.conflict.new_config['name']
-            self.validate(new_name)
-            self.conflict.new_config['name'] = new_name
-
-        def _validate(self, new_name):
-            """Validate new_name is not in database for current user"""
-            if new_name is None:
-                raise ValueError("No new name provided. Cannot resolve experiment name conflict.")
-
+            self.old_name = self.conflict.old_config['name']
+            self.old_version = self.conflict.old_config.get('version', 1)
+            self.new_version = self.old_version
+            self.validate()
+            self.conflict.new_config['name'] = self.new_name
+            self.conflict.new_config['version'] = self.new_version
+
+        def _validate(self):
+            """Validate new_name is not in database with a direct child for current user"""
             # TODO: WARNING!!! _name_is_unique could lead to race conditions,
             # The resolution may become invalid before the branching experiment is
             # registered. What should we do in such case?
-            if not self._name_is_unique(new_name):
+            if self.new_name != self.old_name:
+                # If we are trying to actually branch from experiment
+                if not self._name_is_unique():
+                    raise ValueError(
+                        "Cannot branch from {} with name {} since it already exists.".format(
+                            self.old_name, self.new_name))
+                # Since the name changes, we reset the version count.
+                self.new_version = 1
+
+            # If the new name is the same as the old name, we are trying to increment
+            # the version of the experiment.
+            elif self._check_for_children():
                 raise ValueError(
                     "Experiment name \'{0}\' already exist for user \'{1}\'".format(
-                        new_name, self.conflict.username))
+                        self.new_name, self.conflict.username))
```

Comment:
`and has children. Version cannot be auto-incremented and a new name is required for branching.